### DISCUSSION TITLE
How to convert renormalized 2nd order force-constant in "xml" format into shengbte supported symmetrized FORCE_CONSTANT

### DISCUSSION THREAD

**Discussion by sc696** [ttadano/alamode#231] [General]
Hello everyone,

Can anyone suggest or provide a script to convert the renormalized 2nd order force-constant in "xml" format into shengbte supported symmetrized FORCE_CONSTANT format. I know that a sample script exists in #192, however that script converts the .xml file into a phonopy FORCE_CONSTANT in compact form. However, to use the same FORCE_CONSTANT in Shengbte one need the full FORCE_CONSTANT, which cannot be generated without the FORCE_SET file. Your help, therefore, is highly requested to solve this issue.

Thanks in advance.

**Comment by ttadano** [comment]
Hi,

I've described a solution to this in #192.

**Comment by dengmou123** [comment]
@sc696 @ttadano 
Hello, friend! Could you please extract the second-order force constant values from the.xml file generated by alamode and convert them into the format of FORCE_CONSTANTS used in phonopy? And the.xml file only contains the simple form of the physical constants; However, it does not contain the complete form of FORCE_CONSTANTS; How to solve it
Thank you very much.

**Reply by ttadano** [reply]
Please use https://github.com/ttadano/alamode/blob/develop/tools/convert_fc2.py with `--full` option.

**Reply by dengmou123** [reply]
@ttadano 
Dr. Tadano, hello; I am using this foot-print to extract the renormalized second-order force constant in BaTiO3; And it is named "FORCE_CONSTANTS"; However, the calculated phonon dispersion relation shows obvious imaginary frequencies. However, when using the MODE = phonons function of AlamoDE to draw the phonon spectrum, the resulting phonon dispersion relation is normal. So I suspect that there might be some problems with the script.
